Scripts/RemediationScripts/Remediate-ConfigureAADAuthExtOnLinuxVMSS.ps1

```diff
@@ -0,0 +1,1109 @@
+                        Write-Host $([Constants]::SingleDashLine)
+                        $NonCompliantVMSSDetails += $VMSS
+                    }
+                    
+                }
+                else
+                {
+                    if($VMSS.OrchestrationMode -eq "Flexible")
+                    {
+                        Write-Host "Virtual Machine Scale Set [$($VMSS.ResourceName)] Orchestration type is Flexible. Skipping..." -ForegroundColor $([Constants]::MessageType.Warning)
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+                    else
+                    {
+                        Write-Host "Virtual Machine Scale Set [$($VMSS.ResourceName)] OS type is Windows. Skipping..." -ForegroundColor $([Constants]::MessageType.Warning)
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+                }
+            }
+
+        }
+        catch
+        {
+            Write-Host "Error fetching Virtual Machine Scale Set(s) from the subscription. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+        }    
+    }
+    else
+    {
+        if (-not (Test-Path -Path $FilePath))
+        {
+            Write-Host "ERROR: Input file - $($FilePath) not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+        Write-Host "Fetching all Virtual Machine Scale Set(s) from [$($FilePath)]..." 
+
+        $VirtualMachineResources = Import-Csv -LiteralPath $FilePath
+        $validVirtualMachineResources = $VirtualMachineResources| Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+      
+        $validVirtualMachineResources| ForEach-Object {
+        $resourceId = $_.ResourceId
+            try
+            {
+                $VirtualMachineResource =  Get-AzVM -ResourceGroupName $_.ResourceGroupName -Name $_.ResourceName -ErrorAction SilentlyContinue
+            
+                $VMSSDetails += $VirtualMachineResource  | Select-Object @{N='ResourceId';E={$_.Id}},
+                                                                            @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                                            @{N='ResourceName';E={$_.Name}},
+                                                                            @{N='OSType';E={$_.StorageProfile.OsDisk.OSType}} 
+            }                                                      
+            catch
+            {
+                Write-Host "Error fetching Virtual Machine Scale Set(s) resource: Resource ID - $($resourceId). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            }
+        }
+        Write-Host "Validating and Filtering Linux Virtual Machine Scale Set(s):" -ForegroundColor $([Constants]::MessageType.Info)
```

Comment:
Keep message consistent in both file vs direct run mode.

Reply:
Aligned